docs/admin/rescheduler.md

```diff
@@ -0,0 +1,51 @@
+---
+assignees:
+- davidopp
+- filipg
+- piosz
+
+---
+
+* TOC
+{:toc}
+
+## Critical addons
+
+In addition to Kubernetes core components like api-server, scheduler, controller-manager running on a master machine
+there is a bunch of addons which due to various reasons have to run on a regular cluster node, not the master.
+Some of them are critical to have fully functional cluster: Heapster, DNS, UI.
+Users can break their cluster by evicting a critical addon (either manually or as a side effect of an other operation like upgrade)
+which possibly can become pending (for example when the cluster is highly utilized).
+
+## Rescheduler: guaranteed scheduling of critical addons
+
+Rescheduler ensures that critical addons are always scheduled (assuming the cluster is big enough).
+If one of them is marked by scheduler as unschedulable (pod condition `PodScheduled` set to `false`,
```

Comment:
I would change this to "If the scheduler determines that no node has enough free resources to run the critical addon pod given the pods that are already running in the cluster (indicated by critical addon pod's pod condition `PodScheduled` set to `false`, the reason set to `Unschedulable`) the rescheduler tries to free up space for the addon by evicting some pods; then the scheduler will schedule the addon pod.